src/firefly/java/edu/caltech/ipac/table/io/FITSTableReader.java

```diff
@@ -434,9 +427,14 @@ public static DataType convertToDataType(StarTable table, int colIdx, String str
                     "Unrecognized format character in FITS table file: " + classType);
         }
 
-        if (Objects.equals(strategy, DEFAULT) && colInfo.isArray()) {
-            java_class = String.class;
-            dataType.setTypeDesc(DataType.LONG_STRING);
+        if (colInfo.isArray()) {
+            int[] shape = colInfo.getShape();
+            if (shape != null) {
+                String arraySize = Arrays.stream(shape)
+                                        .mapToObj(d -> d > 0 ? d+"" : "*")
+                                        .collect(Collectors.joining("x"));
+                dataType.setArraySize(arraySize);
```

Comment:
I am having trouble understand what this does. Looks like you are building the array into a string but not keeping values less than 0?  I am also confused why the array data  be put in a object that is `setArraySize`

Reply:
Because array type can be multi-dimensional, `arraySize` is used to describe the shape of the array.  We're using VOTable's `arraysize` convention:  http://www.ivoa.net/documents/VOTable/20190403/WD-VOTable-1.4-20190403.html#ToC12

Basically, `D1xD2xDn` where `D` is a positive integer unless it's the last one, in which case it may be, `*`.

The above code converts Starlink's shape(int[]) into VOTable's `arraysize`.

Reply:
I think it would be helpful to add this info as a JavaDoc to setArraySize method of DataType. It puzzled me too.

Reply:
Thanks.

Reply:
@tgoldina I did.  It's in the variable's declaration.  
edu/caltech/ipac/table/DataType.java:84

Reply:
Indeed. :)